arbisoft/arbisoft/spiders/jobs_spider.py

```diff
@@ -0,0 +1,55 @@
+import scrapy
+import datetime
+
+
+class JobsSpider(scrapy.Spider):
+    name = 'jobs'
+    start_urls = [
+        'https://news.ycombinator.com/jobs'
+    ]
+
+    def parse(self, response):
+
+        items = response.css("tr.athing, td.subtext")
+        all_job_titles = []
+        all_company_urls = []
+        all_job_urls = []
+        all_job_posting_dates = []
+        for item in items:
+
+            job_title = item.css("td.title a.storylink::text").get()
+            company_url = item.css("td.title span.sitebit.comhead a span.sitestr::text").get()
+            job_url = item.css("td.title a.storylink ::attr(href)").get()
+            job_posting_date = item.css("span.age a::text").get()
+
+            if job_title is not None:
+                if 'hiring ' in job_title:
+                    job_title = job_title.split("hiring ")[1]
+                    if job_title.startswith('– '):
+                        job_title = job_title.replace('– ', '')
+                elif 'Hiring ' in job_title:
+                    job_title = job_title.split("Hiring ")[1]
+                    if job_title.startswith('– '):
+                        job_title = job_title.replace('– ', '')
```

Comment:
Look into str.lower() method. It will help eliminate some code here.

Reply:
yes I was thinking about that but the problem is I'll have to undo the changes done by lower() after splitting

Reply:
resolved